### PR TITLE
minor: make AuthMechanism non-exhaustive

### PR DESCRIPTION
**Pull request by saghm**
I realized right after merging RUST-359 that I forgot to make the change that was the whole reason we delayed merging it in the first place, so here it is

### CODE REVIEW

**Comment by saghm** [comment]
I've added a doc comment. Interestingly, I couldn't find any official documentation to link to for it; it might not be available until 4.4 is GA

**Comment by patrickfreed** [comment]
Sorry one more nit, can we include that it is only supported in 4.4+?

**Comment by saghm** [comment]
Oops, good catch; updated

File: src/client/auth/mod.rs [hunk 1]
```diff
@@ -75,6 +76,13 @@ pub enum AuthMechanism {
     /// Note: This mechanism is not currently supported by this driver but will be in the future.
     Plain,
 
+    /// MONGODB-AWS authenticates using AWS IAM credentials (an access key ID and a secret access
+    /// key), temporary AWS IAM credentials obtained from an AWS Security Token Service (STS)
+    /// Assume Role request, or temporary AWS IAM credentials assigned to an EC2 instance or ECS
+    /// task.
+    ///
+    /// Note: Only server versions 4.4+ suppport AWS authentication. Additionally, the driver only
```
Comment (isabelatkinson) [inline src/client/auth/mod.rs:84]: nit: there's an extra p in support

Comment (saghm) [reply src/client/auth/mod.rs]: Oops, not sure how that happened! Fixed

**Pull request merged**